gamemodes/terrortown/gamemode/client/cl_tbuttons.lua

```diff
@@ -163,25 +167,36 @@ function TBHUD:Draw(client)
 		pos = ent:GetPos()
 		scrpos = pos:ToScreen()
 
-		if util.IsOffScreen(scrpos) or not ent:IsUsable() then continue end
+		local percent = math.Clamp((ent:GetNextUseTime() - CurTime()) / ent:GetDelay(), 0, 1)
+		local end_angle = 360 * percent
+
+		-- if util.IsOffScreen(scrpos) or not ent:IsUsable() then continue end
 
 		local usableRange = ent:GetUsableRange()
 
-		if not val.access and not showToAdmins then continue end
+		if not val.visible and not val.access and not showToAdmins then continue end
 
 		d = pos - plypos
 		d = d:Dot(d) / (usableRange * usableRange)
 
 		-- draw if this button is within range, with alpha based on distance
 		if d >= 1 then continue end
 
+		local alpha = 200 * (1 - d)
+		local outlineColorAlpha = ColorAlpha(outlineColor, alpha)
+
+		if ent:GetDelay() > 0 and percent > 0 then
+			draw.Arc(ent:EntIndex(), scrpos.x, scrpos.y, circle_radius, outline_width, 0, 360, 0, ColorAlpha(COLOR_BLACK, alpha))
```

Comment:
FYI, that won't work because these IDs might already be used. I'd do it like this:

```lua
ent.arcCacheID = draw.Arc(ent.arcCacheID, scrpos.x, scrpos.y, circle_radius, outline_width, 0, 360, 0, ColorAlpha(COLOR_BLACK, alpha))
```

Reply:
Worst case the arc is recalculated isnt it? So i would not really worry about it

Reply:
i changed it anyway, doesn't really hurt for now

Reply:
should still be resolved